x/concentrated-liquidity/tick.go

```diff
@@ -253,6 +253,84 @@ func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]quer
 	return liquidityDepthsForRange, nil
 }
 
+// GetLiquidityNetInDirection returns an array of liquidity depth in the given zeroForOne direction.
+// Uses boundTick if given, and iterates until we hit bound Tick's index. If not, uses min tick || max tick depending on the direction of zeroForOne.
+func (k Keeper) GetLiquidityNetInDirection(ctx sdk.Context, poolId uint64, tokenIn string, boundTick sdk.Int) ([]query.LiquidityDepth, error) {
+	// check if pool exists
+	if !k.poolExists(ctx, poolId) {
+		return []query.LiquidityDepth{}, types.PoolNotFoundError{PoolId: poolId}
+	}
+
+	// get min and max tick for the pool
+	p, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return []query.LiquidityDepth{}, err
+	}
+	currentTick := p.GetCurrentTick()
+
+	// sanity check that given tokenIn is an asset in pool.
+	if tokenIn != p.GetToken0() && tokenIn != p.GetToken1() {
+		return []query.LiquidityDepth{}, types.TokenInDenomNotInPoolError{TokenInDenom: tokenIn}
+	}
+
+	// figure out zero for one depending on the token in.
+	zeroForOne := p.GetToken0() == tokenIn
+
+	// use max or min tick if provided bound is nil
+	// var boundTick sdk.Int
+	if boundTick.IsNil() {
+		exponentAtPriceOne := p.GetPrecisionFactorAtPriceOne()
+		minTick, maxTick := math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
+
+		if zeroForOne {
+			boundTick = sdk.NewInt(minTick)
+		} else {
+			boundTick = sdk.NewInt(maxTick)
+		}
+	} else { // if bound is not nil, sanity check that given bound is in the direction of swap
+		if (zeroForOne && boundTick.GT(currentTick)) || (!zeroForOne && boundTick.LT(currentTick)) {
+			return []query.LiquidityDepth{}, types.InvalidDirectionError{ZeroForOne: zeroForOne, PoolTick: currentTick.Int64(), TargetTick: boundTick.Int64()}
+		}
+	}
+
+	liquidityDepths := []query.LiquidityDepth{}
+	swapStrategy := swapstrategy.New(zeroForOne, sdk.ZeroDec(), k.storeKey, sdk.ZeroDec())
+
+	checkTickWithInBoundFn := func(zeroForOne bool, currentTick, boundTick sdk.Int) bool {
+		return (currentTick.GTE(boundTick) && zeroForOne) || (currentTick.LTE(boundTick) && !zeroForOne)
+	}
+
+	nextTick, ok := swapStrategy.NextInitializedTick(ctx, poolId, currentTick.Int64())
+	if !ok {
+		return []query.LiquidityDepth{}, nil
+	}
+	for checkTickWithInBoundFn(zeroForOne, nextTick, boundTick) {
+		tick, err := k.getTickByTickIndex(ctx, poolId, nextTick)
+		if err != nil {
+			return []query.LiquidityDepth{}, err
+		}
+
+		liquidityDepth := query.LiquidityDepth{
+			LiquidityNet: tick.LiquidityNet,
+			TickIndex:    nextTick,
+		}
+		liquidityDepths = append(liquidityDepths, liquidityDepth)
+		currentTick = nextTick
+
+		nextInitTick, ok := swapStrategy.NextInitializedTick(ctx, poolId, currentTick.Int64())
+		// break and return the liquidity as is if
+		// - there are no more next tick that is initialized,
+		// - we hit upper limit
+
+		if !ok {
+			break
+		}
+		nextTick = nextInitTick
+	}
+
+	return liquidityDepths, nil
```

Comment:
We also need to return the active tick liquidity as a separate field

Reply:
IMO current liquidity at index 0 SGTM

Reply:
On more thinking, we currently return the following structure:
```
query.LiquidityDepth{
	LiquidityNet: tick.LiquidityNet,
	TickIndex:    nextTick,
}
```

Current liquidity is semantically incompatible with it because it is not "liquidity net". So it might be better to return it separately so that we don't confuse current liquidity with "liquidity net".

Reply:
Added field for current pool liquidity